semver/increment.ts

```diff
@@ -107,7 +107,7 @@ export function increment(
     // If the input is a non-prerelease version, this acts the same as
     // prepatch.
     case "prerelease":
-      if (version.prerelease.length === 0) {
+      if (version.prerelease?.length === 0) {
```

Comment:
When `prerelease` is `undefined`, that should be considered as the same as empty array. So this line seems introducing a bug. (this passes with prerelease = `[]`, but doesn't pass with prerelease = `undefined`)

Can you add test cases to `increment_test.ts` to exercise these patterns?

Reply:
Probably we need to translate `version.prerelease.length === 0` to `version.prerelease === undefiend || version.prerelease.length === 0`

Reply:
done